pkg/token/identitytoken/identitytoken.go

```diff
@@ -95,8 +56,8 @@ func (c *Client) GenerateToken(ctx context.Context, tenantID, clientID, clientSe
 	}
 	req.Header.Set("Content-Type", "application/json")
 
-	resp, err := doRetries(func() (*http.Response, error) {
-		return c.httpClient.Do(req)
+	resp, err := tokenutil.DoRetries(func() (*http.Response, error) {
+		return httpClient.Do(req)
```

Comment:
You should be able to move the switch statement below to a common function too.

Reply:
The switch block is now put in tokenutil package under the func **ManageHTTPErrorCodes**.
I also found that the error 401/403 were manage by the same case in issuertoken so I added another test for issuertoken to manage 403 error code.